pyiceberg/table/__init__.py

```diff
@@ -760,7 +852,10 @@ def update_table_metadata(base_metadata: TableMetadata, updates: Tuple[TableUpda
     for update in updates:
         new_metadata = _apply_table_update(update, new_metadata, context)
 
-    return new_metadata.model_copy(deep=True)
+    if enforce_validation:
+        return TableMetadataUtil.parse_obj(new_metadata.model_dump())
```

Comment:
This is probably quite expensive, but I think it is worth paying.

Reply:
👍 Since for create-table-transaction commit, we are generating the entire metadata model via a chain of `model_copy` without any validation, I feel it safer to trigger a validation on the model before return :D